homeassistant/components/zimi/__init__.py

```diff
@@ -0,0 +1,64 @@
+"""The zcc integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import CONTROLLER, DOMAIN
+from .controller import ZimiController
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Connect to Zimi Controller and register device."""
+    _LOGGER.debug("Zimi setup starting")
+
+    controller = ZimiController(hass, entry)
+    connected = await controller.connect()
+    if not connected:
+        return False
+
+    hass.data[CONTROLLER] = controller
```

Comment:
use `entry.rntime_data` instead

Reply:
Updated in https://github.com/markhannon/core-integration-zimi/commit/280f851318a05583f5bf479b49259939ae185743

@mhannon11 We need to make this same change for all the other entities as we bring them over